### PR TITLE
support setting raft server init options and raft callback in raft_launcher::init()

### PR DESCRIPTION
**Pull request by sheepgrass**
support setting raft server init options and raft callback in raft_launcher::init()

### CODE REVIEW

File: include/libnuraft/launcher.hxx [hunk 1]
```diff
@@ -45,7 +47,9 @@ public:
                           ptr<logger> lg,
                           int port_number,
                           const asio_service::options& asio_options,
-                          const raft_params& params);
+                          const raft_params& params,
+                          const raft_server::init_options opt = raft_server::init_options(),
```
Comment (greensky00) [inline include/libnuraft/launcher.hxx:51]: Could you please use reference?
```C++
const raft_server::init_options& opt = raft_server::init_options()
```

File: src/launcher.cxx [hunk 2]
```diff
@@ -32,7 +32,9 @@ ptr<raft_server> raft_launcher::init(ptr<state_machine> sm,
                                      ptr<logger> lg,
                                      int port_number,
                                      const asio_service::options& asio_options,
-                                     const raft_params& params_given)
+                                     const raft_params& params_given,
+                                     const raft_server::init_options opt,
+                                     cb_func::func_type raft_callback)
```
Comment (greensky00) [inline src/launcher.cxx:37]: As I mentioned in the previous PR, I'd like to suggest you to move `cb_func::func_type raft_callback` inside `raft_server::init_options` like this:
```C++
struct init_options {
    // ...

    bool skip_initial_election_timeout_;
    
    cb_func::func_type raft_callback_;
};
```
and invoke `ctx->set_cb_func(...)` is inside the constructor of `raft_server`. Then launcher doesn't need to call `set_cb_func`, thus we can get benefit for both launcher and raft_server. What do you think?

**Review by greensky00 — COMMENTED** [review]
Thanks @sheepgrass , some minor comments.

File: include/libnuraft/launcher.hxx (hunk 1 above)
Comment (sheepgrass) [reply include/libnuraft/launcher.hxx:51]: Agree.

Actually, I was just copied from your unit test case :p
https://github.com/eBay/NuRaft/blob/75017885cd0ad5f10c805f035fa5b44ae991aa3a/tests/unit/raft_package_fake.hxx#L51-L54

Comment (greensky00) [reply include/libnuraft/launcher.hxx:51]: Oh, good catch, it also needs to be fixed. Thanks :)

File: src/launcher.cxx (hunk 2 above)
Comment (sheepgrass) [reply src/launcher.cxx:37]: Sorry that I didn't notice you mentioned about moving it to raft_server constructor.  I will make the change.

Actually, for this change, I am also taken reference to your unit test case:
https://github.com/eBay/NuRaft/blob/75017885cd0ad5f10c805f035fa5b44ae991aa3a/tests/unit/raft_package_fake.hxx#L85-L90

**Review by greensky00 — APPROVED** [review]

**Pull request merged**